<?xml version="1.0"?>
<robot name="slider" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:include filename="$(find planar_folding_assembly)/urdf/constants.urdf.xacro"/>

  <xacro:property name="slider_w" value="0.02" />
  <xacro:property name="slider_l" value="0.1" />

  <xacro:property name="handle_distance" value="0.01" />

  <xacro:property name="slider_box_l" value="${slider_l - slider_w/2}" />

  <xacro:arg name="model_plugin" default="libgazebo_ros_empty_controller.so" />

  <link name="slider_handle" >

    <collision>
      <geometry>
        <box size="${handle_size} ${handle_size} ${handle_size}" />
      </geometry>
    </collision>

    <visual>
      <geometry>
        <box size="${handle_size} ${handle_size} ${handle_size}" />
      </geometry>
      <material name="orange"/>
    </visual>

    <inertial>
      <mass value="${handle_mass}"/>
      <inertia
        ixx="1.0" ixy="0.0" ixz="0.0"
        iyy="1.0" iyz="0.0"
        izz="1.0"/>
    </inertial>

  </link>

  <joint name="slider_joint" type="revolute">

    <origin xyz="0 0 ${-handle_distance-handle_size/2}" rpy="0 0 0" />
    <parent link="slider_handle" />
    <child link="slider" />
    <limit lower="0" upper="0" velocity="1" effort="1" />

  </joint>

  <link name="slider">

    <collision>
      <origin xyz="0 0 ${-slider_box_l/2}" rpy="0 0 0" />
      <geometry>
        <box size="${slider_w} ${objects_depth} ${slider_box_l}" />
      </geometry>
    </collision>
    <collision>
      <origin xyz="0 0 ${-slider_box_l}" rpy="${pi/2} 0 0" />
      <geometry>
        <cylinder radius="${slider_w/2}" length="${objects_depth}" />
      </geometry>
    </collision>

    <visual>
      <origin xyz="0 0 ${-slider_box_l/2}" rpy="0 0 0" />
      <geometry>
        <box size="${slider_w} ${objects_depth} ${slider_box_l}" />
      </geometry>
    </visual>
    <visual>
      <origin xyz="0 0 ${-slider_box_l}" rpy="${pi/2} 0 0" />
      <geometry>
        <cylinder radius="${slider_w/2}" length="${objects_depth}" />
      </geometry>
    </visual>

    <inertial>
      <origin xyz="0 0 ${-slider_l/2}" rpy="0 0 0"/>
      <mass value="1"/>
      <inertia
        ixx="1.0" ixy="0.0" ixz="0.0"
        iyy="1.0" iyz="0.0"
        izz="1.0"/>
    </inertial>

  </link>

</robot>
